File: array/160.cpp
#include <bits/stdc++.h>

using namespace std;

using vi = vector<int>;
vi findErrorNums(vector<int>& nums) {
    vi freq(nums.size() + 1, 0);
    vi answer;
    int foundRepeatedElem = 0, missingElem = 0;

    for(const auto & value: nums)
        freq[value]++;

    for(int i = 1; i <= freq.size(); ++i) {
        if(freq[i] == 2)
            foundRepeatedElem = i;
        if(freq[i] == 0)
            missingElem = i;

        if(foundRepeatedElem && missingElem)
            break;
    }

    return {foundRepeatedElem, missingElem};
}

int main() {
    vi elems = {1,1};
    for(auto e: findErrorNums(elems))
        cout<<e<<" ";

    return 0;
}

File: array/160.py
from typing import List


def findErrorNums(nums: List[int]) -> List[int]:
    freq = [0] * (len(nums) + 1)
    answer = [0, 0]

    for elem in nums:
        freq[elem] += 1

    for index in range(1, len(freq)):
        if freq[index] == 2:
            answer[0] = index
        if freq[index] == 0:
            answer[1] = index

    return answer


print(findErrorNums([1, 2, 2, 4]))
